### DISCUSSION TITLE
V5: Validate request multiple times

### DISCUSSION THREAD

**Discussion by vkaelin** [adonisjs/core#1489] [Help] [answered]
Hi! 

Is there a way to validate the incoming request multiple times with different Validators.
I tried the following code but the `translation` variable has the content of the first validation.

I tried to do it this way so that I could use my Validators in another controller.

```ts
public async store ({request, response}: HttpContextContract) {
  const metadata = await request.validate(ArticleStoreValidator)
  const translation = await request.validate(ArticleTranslationStoreValidator)

  const article = await Article.create(metadata)
  await article.related('translations').create(translation)

  await article.preload('translations')

  return response.json(article)
 }
```

Thanks!

**Comment by vkaelin** [comment, marked as answer]
It actually works if I edit the `cacheKey` of my ArticleTranslationStoreValidator Validator.

For example:
```ts
public cacheKey = `${this.ctx.routeKey}-translation`
```

Is it the right thing to do ?

**Reply by RomainLanz** [reply]
Hey @vkaelin! 👋 

Yes this is the right thing to do since both validator have the same `routeKey`.
You can read more about that in the [Schema Caching](https://preview.adonisjs.com/guides/validator/schema-caching) documentation.

**Reply by vkaelin** [reply]
Thanks for the answer!
